File: packages/types/src/interaction/followup-question.ts
/**
 * Follow-up Question Types
 * Types related to follow-up question requests and responses
 */

/**
 * A single question in a follow-up question request
 */
export interface FollowupQuestion {
  /** Question index (0-based) */
  index: number;
  /** The question text */
  text: string;
  /** Available options for the user to choose from */
  options: Array<{
    value: string;
    description?: string;
  }>;
}

/**
 * Structured Follow-up Question Request Data
 * Rich context for follow-up question requests
 */
export interface FollowupQuestionRequestData {
  /** List of questions to ask */
  questions: FollowupQuestion[];
  /** Label for additional information input */
  additionalInfoLabel?: string;
}

/**
 * Structured Follow-up Question Response Data
 * User's response to follow-up question request
 */
export interface FollowupQuestionResponseData {
  /** Answers to the questions */
  answers: Array<{
    questionIndex: number;
    selectedOptionIndex: number; // -1 if custom input
    customInput?: string;
    answer: string;
  }>;
  /** Additional information provided by the user */
  additionalInfo?: string;
}

File: crates/foundation/wf-types/src/interaction/followup_question.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct FollowupQuestion {
    pub index: u32,
    pub text: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub options: Option<Vec<FollowupQuestionOption>>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct FollowupQuestionOption {
    pub label: String,
    pub value: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct FollowupQuestionRequestData {
    pub questions: Vec<FollowupQuestion>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub additional_info_label: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct FollowupQuestionResponseData {
    pub answers: Vec<FollowupAnswer>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub additional_info: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct FollowupAnswer {
    pub question_index: u32,
    pub selected_value: Option<String>,
    pub custom_input: Option<String>,
}
